cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -1029,6 +1033,44 @@ Status LoopPySequence(PyObject* sequence, T func) {
   return Status::OK();
 }
 
+template <typename T>
+Status LoopPySequenceWithMasks(
+    PyObject* sequence,
+    const Ndarray1DIndexer<uint8_t>& mask_values,
+    bool have_mask,
+    T func
+) {
```

Comment:
Can you run clang-format? (`make format` or `ninja format`). This should also fix the cpplint failure in CI . See https://github.com/apache/arrow/tree/master/cpp#continuous-integration